packages/helia/src/utils/networked-storage.ts

```diff
@@ -46,10 +68,11 @@ export class NetworkedStorage implements Blocks {
       return cid
     }
 
-    if (this.bitswap?.isStarted() === true) {
-      options.onProgress?.(new CustomProgressEvent<CID>('blocks:put:bitswap:notify', cid))
-      this.bitswap.notify(cid, block, options)
-    }
+    options.onProgress?.(new CustomProgressEvent<CID>('blocks:put:providers:notify', cid))
+
+    this.blockProviders.forEach(provider => {
+      provider.notify(cid, block, options)
```

Comment:
```suggestion
      provider.notify?.(cid, block, options)
```

or better: keep a list of blockProviders that have `notify` methods and only call those.

Reply:
This method is now optional on the interface.